### ISSUE TITLE
MaterialDesignFloatingHintTextBox and MaterialDesignFloatingHintComboBox are not matching again

### ISSUE THREAD

**Issue by bebenins** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2528]
This was finally fixed a while back but recently got messed up again (at least in 4.3.0.3).  Unfortunately I can't tell exactly when it got messed up.
Top example is 4.3.0.3.
Bottom example is an older release (perfecto!)
![image](https://user-images.githubusercontent.com/16761301/146557764-6baacc06-1233-429a-841b-6543535cec7d.png)

**Comment by MichelMichels** [comment]
Could you upload your code? I don't see this behaviour when I try to replicate.

**Comment by bebenins** [comment]
> Could you upload your code? I don't see this behaviour when I try to replicate.

Yes, sorry, I thought this is global but looks like it only happens when you add a Button to ToolBar
This is the link to a tiny project to reproduce it https://drive.google.com/file/d/1VJgsg-vADZbfwIsr6LPAeFvBChHJHdQC/view?usp=sharing
But if you don't feel like downloading, this is XAML that does it for me:

```<ToolBarTray DockPanel.Dock="Top" ToolBarTray.IsLocked="True">
            <ToolBar Style="{StaticResource MaterialDesignToolBar}">
                <ComboBox materialDesign:HintAssist.Hint="Groups" MaxDropDownHeight="1000" Margin="10,0,10,0" Style="{StaticResource MaterialDesignFloatingHintComboBox}">
                    <ComboBoxItem>Employee</ComboBoxItem>
                    <ComboBoxItem>Contractor</ComboBoxItem>
                    <ComboBoxItem>Supplier</ComboBoxItem>
                </ComboBox>
                <TextBox Text="{Binding SystemSn}" Margin="10,0,10,0" materialDesign:HintAssist.Hint="System Serial #" Style="{StaticResource MaterialDesignFloatingHintTextBox}"/>
                <Button ToolTip="Show All Groups Stats">
                    <materialDesign:PackIcon Kind="Teamviewer"/>
                </Button>
            </ToolBar>
        </ToolBarTray>

**Comment by MichelMichels** [comment]
I pinpointed the problem to the ComboBox style with key `MaterialDesignComboBox`.
It has its `VerticalAlignment` property set at `Stretch`. This should be `Center` to fix the behaviour seen in your issue.

The thing is, the `VerticalAlignment` of `MaterialDesignTextBoxBase` is not set, so I assume it gets a default version of `Center`. This style also has a "VerticalContentAlignment" of "Top" instead of "Stretch" in `MaterialDesignComboBox`.

I'll try to find out why it's explicitly set to `VerticalAlignment="Stretch"` in the ComboBox style.

**Comment by MichelMichels** [comment]
I traced it back to this PR: https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/pull/2428

**Comment by bebenins** [comment]
Thanks guys 👍

**Comment by MichelMichels** [comment]
Fixed with  PR #2534. @bebenins I included a ToolBar like yours in the demo app to showcase the fix.

**Comment by bebenins** [comment]
> Fixed with PR #2534. @bebenins I included a ToolBar like yours in the demo app to showcase the fix.

Thanks 👍
